deploy/sre-build-test/50-build-test.CronJob.yaml

```diff
@@ -0,0 +1,51 @@
+---
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: sre-build-test
+  namespace: openshift-build-test
+spec:
+  failedJobsHistoryLimit: 5
+  successfulJobsHistoryLimit: 3
+  concurrencyPolicy: Forbid
+  schedule: "11 */1 * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          affinity:
+            nodeAffinity:
+              requiredDuringSchedulingIgnoredDuringExecution:
+                nodeSelectorTerms:
+                - matchExpressions:
+                  - key: "node-role.kubernetes.io"
+                    operator: In
+                    values:
+                      - infra
+          tolerations:
+          - operator: Equal
+            key: node-role.kubernetes.io
+            value: infra
+            effect: NoSchedule
+          serviceAccountName: sre-build-test
+          restartPolicy: Never
+          containers:
+          - name: sre-build-test
+            image: image-registry.openshift-image-registry.svc:5000/openshift/cli:latest
+            imagePullPolicy: Always
+            command:
+            - /bin/sh
+            - -ec
```

Comment:
The `-e` option here (which I like) will cause the CronJob to litter namespaces should the build fail for whatever reason prior to line 51. I'm not sure how you want to handle that.

Reply:
Good point, this can be removed.  I want the job to wait forever if there's a problem.  Updating..

Reply:
Is it desirable to wait forever? `Job`s (and by extension `CronJob`s) have an innate retry (by default, it's 6 times) if the thing fails.

I don't know what happens with `CronJob` failure accounting if the wait is very long. I do know that at one point failed `CronJob`s could stop triggering _at all_ after a certain number of failures (I think 100), although that might have been changed, as that was ca. 2017 I ran into it.

Reply:
Set `concurrencyPolicy: Forbid` [1] expecting that to keep it from trying if the job is already running.  I didn't want it to delete the generated test namespace.  If the build is stuck we need it to stay around for the alert to trigger and persist.  If there's some other problem I think we have `KubeJobCompletion` alert that will fire for this job after 1 hour.

[1] https://github.com/openshift/managed-cluster-config/pull/234/files#diff-016fe875d079f89b3ad58cdfcd27de45R10

Reply:
If the build is stuck won't the existing `KubeJobCompletion` rule already cover us?

Reply:
No, I see this alert in a prod cluster for a broken customer build.  There's no other alerts in cluster.  The reason I want to create a new build for a targeted alert is so broken customer builds don't page SRE.

Reply:
I understand.